portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,10 +33,22 @@
 /** Servlet that sends and receives comments data. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private ArrayList<String> comments = new ArrayList();
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> comments = new ArrayList();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String body = (String) entity.getProperty("body");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      comments.add(body);
+    }
+
     Gson gson = new Gson();
     response.setContentType("text/html;");
     response.getWriter().println(gson.toJson(comments));
```

Comment:
Did you have corresponding JS changes to render the new comment object?

Reply:
No, the comment object is still just a string at this point, so it's rendered the same way.